### PR TITLE
[RFC][NI] Add options to map

### PR DESCRIPTION
**Pull request by AndreJoaquim**
# What this does?
- Use new ember lts on ember-try
- Migrates to `eslintrc`
- Upgrades `mapbox.js` version to `3.1.1`
- Upgrades dependencies
- Fixes `rootURL`
- Adds default `mapboxToken`
- Improve `module-for-acceptance.js`

- Add `mapbox-circle`
- Add customisation options to `mapbox-map`
- Add flag to be able to exclude marker cluster
- Improves `mapbox-geojson` so it supports custom styles
- Add `mapbox-layer`
- Implement default `createMarkerIcon` on `mapbox-marker`
- Add events to `mapbox-marker`
- Handles removal of markers in cluster when added dynamically
- Add cluster events
- Create `mapbox-mock`

## TODO
- Fix tests that are now breaking. **Will tackle this during this week**

## Props 👏 
@montoias @AndreJoaquim

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +35,7 @@ corresponding mapId form mapbox.
 If `divId` is not specified it defaults to 'map'.
 
 You can specify the zoom level, center of the map, and trigger an `onclick` event as attributes.
-The `onclick` event will return an instance of the event containing the lat,lng, among other properties.
+The `onclick` event will return an instance of the event containing the lat, lng, among other properties.
```
Comment (mmelo) [inline README.md:38]: Non-blocking but....
The `onclick` event will return an instance of the event containing the lat and lng, among other properties.
or
The `onclick` event will return an instance of the event containing the lat, lng and other properties.

**Review by AndreJoaquim — COMMENTED** [review]
I approve this message!

File: app/components/mapbox-circle.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default } from 'ember-cli-mapbox/components/mapbox-circle';
```
Comment (mmelo) [inline app/components/mapbox-circle.js:1]: newline @ EOF

File: app/components/mapbox-layer.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from 'ember-cli-mapbox/components/mapbox-layer';
```
Comment (mmelo) [inline app/components/mapbox-layer.js:1]: newline @ EOF

**Review by mmelo — APPROVED** [review]
great job 👏 🥇

**Comment by montoias** [comment]
I can review this later today - if you need @AndreJoaquim @mmelo @asantos00

**Comment by asantos00** [comment]
Would be more than welcome @montoias :)

**Comment by migueldemoura** [comment]
We've been using this in prod for a long, long while. Merging so we can iterate over this.

**Pull request merged**